tools/Docgen/Builder/Builder.cs

```diff
@@ -0,0 +1,57 @@
+using System.IO;
+using Docgen.Generators;
+
+namespace Docgen.Builder
+{
+    public static class Builder
+    {
+        public static void Build(DoxygenDatabase database)
+        {
+            CleanOutputDir(Options.Instance.OutputDirectory);
+            BuildDocs(database, Options.Instance.OutputDirectory);
+            BuildTableOfContents(database, Options.Instance.OutputDirectory);
+        }
+
+        private static void BuildTableOfContents(DoxygenDatabase database, string outputDirectory)
+        {
+            // Build the table of contents at {{outputDirectory}}/.toc/toc.md
+            var filepath = Path.Combine(outputDirectory, ".toc", "toc.md");
+            Directory.CreateDirectory(Path.GetDirectoryName(filepath));
+            File.WriteAllText(filepath, new TableOfContentsGenerator(database).Generate());
+        }
+
+        private static void BuildDocs(DoxygenDatabase database, string outputDirectory)
+        {            
+            foreach (var details in database.TypeDetails)
```

Comment:
those loops could potentially be factored out into a method.

Reply:
Yeah agreed.